components/cloud/src/error.rs

```diff
@@ -0,0 +1,126 @@
+use std::fmt::{Debug, Display};
+use std::io::{Error as IoError, ErrorKind};
+use std::{error, result};
+
+use protobuf::ProtobufError;
+
+use error_code::{self, ErrorCode, ErrorCodeExt};
+use tikv_util::stream::RetryError;
+
+pub type Result<T> = result::Result<T, Error>;
+
+pub trait ErrorTrait: Debug + Display + RetryError + Send + Sync + 'static {}
+
+/// The error type for the cloud.
+#[derive(Debug, Fail)]
+pub enum Error {
+    #[fail(display = "Other error {}", _0)]
+    Other(Box<dyn error::Error + Sync + Send>),
+    #[fail(display = "IO error {}", _0)]
+    Io(IoError),
+    #[fail(display = "Protobuf error {}", _0)]
+    Proto(ProtobufError),
+    #[fail(display = "API Timeout error: {}", _0)]
+    ApiTimeout(Box<dyn error::Error + Sync + Send>),
+    #[fail(display = "API internal error: {}", _0)]
+    ApiInternal(Box<dyn error::Error + Sync + Send>),
+    #[fail(display = "Key error: {}", _0)]
+    KmsError(KmsError),
+}
+
+impl Error {
+    pub fn boxed(self) -> Box<dyn ErrorTrait> {
+        Box::new(self) as Box<dyn ErrorTrait>
+    }
+}
+
+impl ErrorTrait for Error {}
+
+#[derive(Debug, Fail)]
+pub enum KmsError {
+    #[fail(display = "Wrong master key {}", _0)]
+    WrongMasterKey(Box<dyn error::Error + Sync + Send>),
+    #[fail(display = "Empty key {}", _0)]
+    EmptyKey(String),
```

Comment:
where is this error type being used?

Reply:
This is used by `empty_key_contents`. I can get rid of that helper function.